create spacer chip

bryan-stewart opened the pull request:
## Description

![IMG_5433](https://user-images.githubusercontent.com/111661559/230185311-08614453-738c-4b1f-a3cd-9b2e9ca32b0d.jpg)

I added a spacer chip to the chips card. Just a div element with flex-grow. currently no configuration options but setting custom sizes could be added. 
<!--- Describe your changes in detail -->

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here -->

This PR fixes or closes issue: fixes #1096 

## Motivation and Context

Seems like a good idea to have spacers in the chips card

## How Has This Been Tested

Tested in Firefox, Chrome, Edge, Companion App(iOS)

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

-   [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
-   [x] 🚀 New feature (non-breaking change which adds functionality)
-   [ ] 🌎 Translation (addition or update a translation)
-   [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

-   [x] My code follows the code style of this project.
-   [x] My change requires a change to the documentation.
-   [ ] I have updated the documentation accordingly.
-   [x] I have tested the change locally.

RickeyWard commented on src/cards/chips-card/chips/spacer-chip.ts:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    protected render(): TemplateResult {
+        if (!this.hass || !this._config) {
+            return html`<div></div>`;
+        }
+
+        return html`<div></div>`;
```
This code doesn't seem to have any effect, would probably drop it.

```suggestion
        return html`<div></div>`
```

piitaya replied:
Yes, hass and config are not used, we don't need these attributes in the component.

piitaya commented on src/cards/chips-card/chips/spacer-chip-editor.ts:
```diff
@@ -0,0 +1,55 @@
+        const schema = computeSchema();
+
+        return html`
+            <ha-form
+                .hass=${this.hass}
+                .data=${this._config}
+                .schema=${schema}
+                .computeLabel=${this._computeLabel}
+                @value-changed=${this._valueChanged}
+            ></ha-form>
```
You can delete this because it's not needed.

piitaya commented on src/cards/chips-card/chips/spacer-chip-editor.ts:
```diff
@@ -0,0 +1,55 @@
+        this._config = config;
+    }
+
+    private _computeLabel = (schema: HaFormSchema) => {
+        const customLocalize = setupCustomlocalize(this.hass!);
+
+        if (GENERIC_LABELS.includes(schema.name)) {
+            return customLocalize(`editor.card.generic.${schema.name}`);
+        }
+        return this.hass!.localize(`ui.panel.lovelace.editor.card.generic.${schema.name}`);
+    };
```
You can delete this because it's not needed.

piitaya commented on src/cards/chips-card/chips/spacer-chip-editor.ts:
```diff
@@ -0,0 +1,55 @@
+import { SpacerChipConfig } from "../../../utils/lovelace/chip/types";
+import { LovelaceChipEditor } from "../../../utils/lovelace/types";
+
+const computeSchema = memoizeOne((): HaFormSchema[] => []);
```
You can delete this because it's not needed.

piitaya commented on src/cards/chips-card/chips/spacer-chip-editor.ts:
```diff
@@ -0,0 +1,55 @@
+                .computeLabel=${this._computeLabel}
+                @value-changed=${this._valueChanged}
+            ></ha-form>
+            <div style="margin-top: 20px;">There are currently no config options for this chip.</div>
+        `;
```
This should be translated.

piitaya reviewed (commented):
Hello. I like the idea. It can be very useful.
Thanks's for your contribution. The code need to be cleaned because there is many unused code in the editor and the chip.
Also, the documentation is missing.

bryan-stewart commented:
Sorry, I left the config and form stuff in because I wasn't sure if some config options would be wanted.

I removed the unused code except the setConfig function as it is required on interface LovelaceChip. I'm not sure what is best to do with that.

I added docs and an English translation

bryan-stewart commented:
If I prevent the edit button from rendering for the spacer chip in chips-card-chips-editor then I can get rid of the editor all together. Removes the need for the translation.

<img width="374" alt="Screenshot 2023-04-06 142340" src="https://user-images.githubusercontent.com/111661559/230463583-c9d4276a-6304-4c66-8222-6e5f2d405f83.png">

piitaya commented:
Yes, we can remove the editor and the edit button 😊

alexvaltchev commented:
I updated to the last version but I dont see SPACER available? Why?  [v2.7.0](https://github.com/piitaya/lovelace-mushroom/releases/tag/v2.7.0)
![Screenshot_2](https://user-images.githubusercontent.com/118940240/232711353-2822faf8-34f0-4253-9dd5-1cdb8c31bcb7.jpg)

aviadlevy commented:
Make sure you refresh your browser. working well for me
![image](https://user-images.githubusercontent.com/11833447/232719175-4b8d80cb-fe3b-4aa9-a696-3ca979ce4455.png)

schmuy commented:
> I updated to the last version but I dont see SPACER available? Why?  [v2.7.0](https://github.com/piitaya/lovelace-mushroom/releases/tag/v2.7.0)
> 
> ![Screenshot_2](https://user-images.githubusercontent.com/118940240/232711353-2822faf8-34f0-4253-9dd5-1cdb8c31bcb7.jpg)
> 
> 

Same issues.  I don't see spacer available in safari or iOS companion app.  It does show in chrome and I can add it in chrome and it behaves correctly.  When going back to chrome or iOS it does not work.
